### PR TITLE
allow to enable/disable redirects.

### PR DESCRIPTION
**Pull request by cristifalcas**

### CODE REVIEW

File: templates/server.conf.erb [hunk 1]
```diff
@@ -412,6 +412,9 @@ enabled: <%= scope['pulp::email_enabled'] %>
 #   downloading content from the Squid cache.
 
 [lazy]
+<% unless [nil, :undefined, :undef, ''].include?(scope.lookupvar('pulp::lazy_enabled')) -%>
```
Comment (ekohl) [inline templates/server.conf.erb]: Since this is validated as a bool this check isn't needed.

File: manifests/init.pp [hunk 2]
```diff
@@ -176,6 +176,9 @@
 #
 # $manage_squid::               boolean controls whether squid configuration is managed or manual
 #
+# $lazy_enable::                This controls whether redirect is enabled
```
Comment (ekohl) [inline manifests/init.pp]: To me it's a bit confusing that first it's lazy, then it's redirect. You also have ```$lazy_enable``` where the code has ```$lazy_enabled``` (note the missing d)

File: templates/server.conf.erb (hunk 1 above)
Comment (cristifalcas) [inline templates/server.conf.erb]: This is more for undefined values.
Should I remove it?

File: manifests/init.pp (hunk 2 above)
Comment (cristifalcas) [inline manifests/init.pp]: The explanation is from server.conf:

# enabled:
#   This controls whether redirect is enabled

File: templates/server.conf.erb (hunk 1 above)
Comment (ekohl) [inline templates/server.conf.erb]: Yes. We know it's never any of those values.

**Comment by ehelms** [comment]
I asked the Pulp team about this config option and they indicated they were not aware of it. Can you point out where it comes from?

**Comment by ehelms** [comment]
I should also say I checked https://github.com/pulp/pulp/blob/master/server/etc/pulp/server.conf#L391 first and when I didn't see it in their official server.conf I reached out to them.

**Comment by cristifalcas** [comment]
It was based on the parameter in the server.conf template from the puppet module. If it doesn't exists, can the lazy streaming be disabled somehow?

**Comment by cristifalcas** [comment]
ok, it seems it's a pulp repo parameter.

**Pull request closed without merging**